### PR TITLE
Add Redux

### PR DESCRIPTION
**Pull request by nbagramov**

### CODE REVIEW

File: src/store/reducer.ts [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+      return [...newPurchaseList];
+    }
+    case EDIT_PURCHASE: {
+      const newPurchaseList = [...state];
+      const {id} = (action as EditPurchaseAction).payload;
+      const purchaseIndex = state.findIndex((item) => item.id === id);
+      if (purchaseIndex != -1) {
+        newPurchaseList[purchaseIndex].isEdit = !newPurchaseList[purchaseIndex].isEdit;
+      }
+      return [...newPurchaseList];
```
Comment (IslamRustam) [inline src/store/reducer.ts]: if purchaseIndex **!==** -1 
You shouldn't call `return [...newPurchaseList]` if purchaseIndex === -1. In this case you should just call `return state;`

File: src/store/reducer.ts [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    }
+    case CHANGE_PLACE: {
+      const changePlacePurchaseList = [...state];
+      const {id, place} = (action as ChangePlaceAction).payload;
+      const purchaseIndex = state.findIndex((item) => item.id === id);
+      if (purchaseIndex != -1 && place) {
+        changePlacePurchaseList[purchaseIndex].place = place;
+      }
+      return [...changePlacePurchaseList];
```
Comment (IslamRustam) [inline src/store/reducer.ts]: same as above

File: src/store/reducer.ts [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    }
+    case CHANGE_PRICE: {
+      const newPurchaseList = [...state];
+      const {id, price} = (action as ChangePriceAction).payload;
+      const purchaseIndex = state.findIndex((item) => item.id === id);
+      if (purchaseIndex != -1 && (price || price == 0)) {
+        newPurchaseList[purchaseIndex].price = price;
+      }
+      return [...newPurchaseList];
```
Comment (IslamRustam) [inline src/store/reducer.ts]: same as above

Comment (IslamRustam) [reply src/store/reducer.ts]: and you should really have a reason to use not strict comparison (==) instead of strict comparison (===)

**Review by IslamRustam — CHANGES_REQUESTED** [review]

File: src/store/reducer.ts [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+        purchases: [...newPurchaseList],
+        error: false
+      };
+    }
+    case CREATE_PURCHASE_FAIL: {
+      const newPurchaseList = [...state.purchases];
+      return {
+        purchases: [...newPurchaseList],
+        error: true
+      };
```
Comment (IslamRustamov1) [inline src/store/reducer.ts]: Why do you recreate purchases list? Just error: true is enought

File: src/store/reducer.ts [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+      }
+      return state;
+    }
+    case EDIT_PURCHASE_FAIL: {
+      const newPurchaseList = [...state.purchases];
+      return {
+        purchases: [...newPurchaseList],
+        error: true
+      };
+    }
```
Comment (IslamRustamov1) [inline src/store/reducer.ts]: same as above

File: src/store/reducer.ts [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+        purchases: [...newPurchaseList],
+        error: false
+      };
+    }
+    case DELETE_PURCHASE_FAIL: {
+      const newPurchaseList = [...state.purchases];
+      return {
+        purchases: [...newPurchaseList],
+        error: true
+      };
```
Comment (IslamRustamov1) [inline src/store/reducer.ts]: same as above

File: src/store/reducer.ts [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+        };
+      }
+      return state;
+    }
+    case GET_PURCHASES_FAIL: {
+      const newPurchaseList = [...state.purchases];
+      return {
+        purchases: [...newPurchaseList],
+        error: true
+      };
```
Comment (IslamRustamov1) [inline src/store/reducer.ts]: same as above

**Review by IslamRustamov1 — CHANGES_REQUESTED** [review]

**Pull request merged**